### PR TITLE
[AWS] Add userTags workflow and jobs

### PR DESCRIPTION
**Pull request by yunjiang29**
Adding user tags jobs to:
* 4.12, amd/arm, stable/nightly
* 4.13/4.14, amd/arm, nightly
* 4.12/4.13 stable to 4.13 nightly, amd/arm
* 4.12 stable to 4.13 stable, amd/arm

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-commands.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+
+# Machines
+err_output=$(mktemp)
+machine_output=$(mktemp)
+oc get machine -n openshift-machine-api -owide --no-headers 1>${machine_output} 2>${err_output}
```
Comment (jinyunma) [inline ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-commands.sh:19]: Does this script only use to check master node name? If so, how about to only filter master node via below command?
```
oc get machine -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers
```

Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-commands.sh:19]: OK, I will udpate, thanks

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/37733/rehearse-37733-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28/1641093249282084864/artifacts/aws-ipi-usertags-p2-f28/cucushift-installer-check-machine-name/build-log.txt

```
control_plane_nodes_count: 3
excepted_count: 3
INFO: All control plane machine names are expected.
Machines:
ci-op-8w1b5vyb-9f6b6-kvqdn-master-0   Running   m6i.xlarge   us-east-1   us-east-1a   21m   ip-10-0-180-207.ec2.internal   aws:///us-east-1a/i-0b0a99239ea817161   running
ci-op-8w1b5vyb-9f6b6-kvqdn-master-1   Running   m6i.xlarge   us-east-1   us-east-1f   21m   ip-10-0-229-3.ec2.internal     aws:///us-east-1f/i-0394107ce91374d5a   running
ci-op-8w1b5vyb-9f6b6-kvqdn-master-2   Running   m6i.xlarge   us-east-1   us-east-1a   21m   ip-10-0-173-74.ec2.internal    aws:///us-east-1a/i-077c1be12b1d923bb   running
```

File: ci-operator/step-registry/ipi/conf/aws/usertags/ipi-conf-aws-usertags-commands.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Create 25 user tags
+# 24 tags in this patch + 1 expirationDate tag in ipi-conf-aws step.
+
+PATCH="${ARTIFACT_DIR}/install-config-usertags.yaml.patch"
+cat <<EOF > ${PATCH}
+platform:
+  aws:
+    userTags:
+      key-length-128-def${str100}${str10}: val-length-256-a${str200}${str40}
+      c: c
```
[the remark this answers is not in the document]
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/usertags/ipi-conf-aws-usertags-commands.sh:25]: Thanks @wking, I didn't notice that before, re-use `USER_TAGS` now.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
@jinyunma @jianlinliu PTAL

File: ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+ref:
+  as: cucushift-installer-check-machine-name
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-ref.yaml:2]: how about rename it to `cucushift-installer-check-master-machine-name`

File: ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-commands.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+control_plane_nodes_count=$(cat "${machine_output}" | wc -l || true)
+excepted_count=$(cat "${machine_output}" | awk '{print $1}' | grep -iE "master-[0-9]{1}$" | wc -l || true)
+
+echo "control_plane_nodes_count: ${control_plane_nodes_count}"
+echo "excepted_count: ${excepted_count}"
+
+if [[ "${excepted_count}" != "${control_plane_nodes_count}" ]]; then
+    echo "ERROR: One or more control plane machine name is not expected."
+    exit 1
+else
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/machine-name/cucushift-installer-check-machine-name-commands.sh:35]: How about add one more guard to judge `excepted_count` or `control_plane_nodes_count` > 0 ?

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-usertags-p2-f28

**Comment by yunjiang29** [comment]
@jianlinliu please review again, it's ready for merge now.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 5]
```diff
@@ -1294,6 +1294,28 @@ tests:
...
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-ipi-workers-marketplace
+- as: aws-ipi-usertags-p2-f28
+  cron: 22 17 22 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      E2E_RUN_TAGS: '@aws-ipi and @network-ovnkubernetes and not @fips'
+      TAG_VERSION: '@4.13'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1304]: Should be 4.14 there

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml]: fixed several similar issue

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml]: @liangxia please review again, thanks

**Pull request merged**